src/core/qgscoordinatetransform.cpp

```diff
@@ -707,6 +720,45 @@ void QgsCoordinateTransform::transformCoords( int numPoints, double *x, double *
   }
 #endif
 
+#if PROJ_VERSION_MAJOR>=6
+
+  if ( actualRes != 0 )
```

Comment:
actualRes will not be reliable for numPoints > 1. you'd probably have to iterate over all transformed points to check their x != HUGE_VAL

Reply:
Why is that? Does proj_errno get reset mid-operation when transforming multiple points?

Reply:
I'm not sure. Error handling in PROJ is super confusing (to me at least)... See https://github.com/OSGeo/PROJ/issues/1856